drivers/pwm/pwm_mcux_ftm.c

```diff
@@ -496,13 +505,64 @@ static const struct pwm_driver_api mcux_ftm_driver_api = {
 #define TO_FTM_PRESCALE_DIVIDE(val) _DO_CONCAT(kFTM_Prescale_Divide_, val)
 
 #ifdef CONFIG_PWM_CAPTURE
+#if IS_EQ(DT_NUM_IRQS(DT_DRV_INST(0)), 1)
+static void mcux_ftm_isr(const struct device *dev)
+{
+	const struct mcux_ftm_config *cfg = dev->config;
+
+	mcux_ftm_irq_handler(dev, 0, cfg->channel_count);
+}
+
 #define FTM_CONFIG_FUNC(n) \
 static void mcux_ftm_config_func_##n(const struct device *dev) \
 { \
 	IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority), \
 		    mcux_ftm_isr, DEVICE_DT_INST_GET(n), 0); \
 	irq_enable(DT_INST_IRQN(n)); \
 }
+#else /* Multiple interrupts */
+#define FTM_ISR_FUNC_NAME(suffix) _DO_CONCAT(mcux_ftm_isr_, suffix)
+#define FTM_ISR_FUNC(chan_start, chan_end) \
+static void mcux_ftm_isr_##chan_start##_##chan_end(const struct device *dev) \
+{ \
+	mcux_ftm_irq_handler(dev, chan_start, chan_end + 1); \
+}
+
+#define FTM_ISR_CONFIG(node_id, prop, idx) \
+do { \
+	IRQ_CONNECT(DT_IRQ_BY_IDX(node_id, idx, irq), \
+		    DT_IRQ_BY_IDX(node_id, idx, priority), \
+		    FTM_ISR_FUNC_NAME(DT_STRING_TOKEN_BY_IDX(node_id, prop, idx)), \
+		    DEVICE_DT_GET(node_id), \
+		    0); \
+	irq_enable(DT_IRQ_BY_IDX(node_id, idx, irq)); \
+} while (false);
+
+#define FTM_CONFIG_FUNC(n) \
+static void mcux_ftm_config_func_##n(const struct device *dev) \
+{ \
+	DT_INST_FOREACH_PROP_ELEM(n, interrupt_names, FTM_ISR_CONFIG) \
+}
+
+#if DT_INST_IRQ_HAS_NAME(0, overflow)
```

Comment:
this only works for one instance, but this driver is multi-instance already, and it can't be known which is instance 0